math/src/fft/fft_inputs.rs

```diff
@@ -0,0 +1,55 @@
+use super::StarkField;
+use crate::FieldElement;
+
+// FFTINPUTS TRAIT
+// ================================================================================================
+
+/// Defines the interface that must be implemented by the input to fft_in_place method.
```

Comment:
The description looks good - but I would keep the `shift_by()` methods we had before. I think they simplifies code it on there places.

Reply:
Then we can use these methods in `interpolate_poly` and `interpolate_poly_with_offset` as follows:

```
pub fn interpolate_poly<B, E>(evaluations: &mut [E], inv_twiddles: &[B])
where
    B: StarkField,
    E: FieldElement<BaseField = B>,
{
    fft_in_place(evaluations, inv_twiddles, 1, 1, 0);
    let inv_length = B::inv((evaluations.len() as u64).into());

    // Use fftinputs shift_by on evaluations.
    FftInputs::shift_by(evaluations, inv_length);
    permute(evaluations);
}

pub fn interpolate_poly_with_offset<B, E>(
    evaluations: &mut [E],
    inv_twiddles: &[B],
    domain_offset: B,
) where
    B: StarkField,
    E: FieldElement<BaseField = B>,
{
    fft_in_place(evaluations, inv_twiddles, 1, 1, 0);
    permute(evaluations);

    let domain_offset = B::inv(domain_offset);
    let offset = B::inv((evaluations.len() as u64).into());

    // Use fftinputs's shift_by_series on evaluations.
    FftInputs::shift_by_series(evaluations, offset, domain_offset);
}

```